src/main/java/org/pac4j/dropwizard/Pac4jConfiguration.java

```diff
@@ -1,7 +1,172 @@
 package org.pac4j.dropwizard;
 
-import io.dropwizard.Configuration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-public interface Pac4jConfiguration<T extends Configuration> {
-    Pac4jFactory getPac4jFactory(T configuration);
+import org.pac4j.core.authorization.authorizer.Authorizer;
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.matching.Matcher;
+import org.pac4j.jax.rs.filter.SecurityFilter;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An interface defining getters for parameters required to configure a {@link SecurityFilter}.
+ */
+public class Pac4jConfiguration {
```

Comment:
Ok, but I don't want it to "build" the configuration at the end, because I want users to be able to modify the config instance from the run() method of the `Application` (without having to use something like `ConfigSingleton` provided by pac4j). As that compliant with the Factory approach or is there an idiomatic way to achieve what I want?